olive/engine/engine.py

```diff
@@ -224,8 +241,68 @@ def run(
             json.dump(results, open(results_path, "w"), indent=4)
             return results
 
+        # TODO by myguo: replace the following for loop using the accelerator and excution provider list when adding
+        # accelerator support
+        footprints = []
+        for i in range(1):
```

Comment:
Please correct me if my understand is not correct: 

the overall idea is the engine will run all same passes with different hardware accelerators. 
Do we have (or will have) any pass that requires a specifical hardware?
Will the engine output best models for all different hardware options or  for each hardware option?

Reply:
Yes, the engine would run pass for different hardware accelerators. Next PR will change all olive pass to accept an accelerator spec which is maintained by engine. 
For each hardware and EP, there will be best models. All of these best models will be returned to user.